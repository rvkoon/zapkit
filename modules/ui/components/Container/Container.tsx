import { ReactNode } from "react";
import cx from "classnames";

export interface IContainerProps {
  isWide?: boolean;
  children: ReactNode;
}

export function Container({
  isWide = false,
  children,
}: IContainerProps): JSX.Element {
  return (
    <div
      className={cx(
        "w-full md:w-container m-auto",
        isWide && "w-containerWide"
      )}
    >
      {children}
    </div>
  );
}
