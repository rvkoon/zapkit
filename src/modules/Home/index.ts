export * from './components';
export * from './helpers';
export * from './hooks';
export * from './services';
export * from './types';
export * from './constants';