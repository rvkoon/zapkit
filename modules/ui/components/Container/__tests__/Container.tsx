import { Container, IContainerProps } from "../Container";
import { RenderResult } from "@testing-library/react";
import { render } from "modules/tests";

describe("Container component", () => {
  function renderComponent(props: IContainerProps): RenderResult {
    return render(<Container {...props} />);
  }

  it("should render correctly", () => {
    const { container } = renderComponent({ children: <div></div> });
    expect(container).toMatchSnapshot();
  });
});
