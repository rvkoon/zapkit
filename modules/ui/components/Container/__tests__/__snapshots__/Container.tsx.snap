// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container component should render correctly 1`] = `
<div>
  <div
    class="w-full md:w-container m-auto"
  >
    <div />
  </div>
</div>
`;
